Framework/Algorithms/test/ScaleXTest.h

```diff
@@ -314,21 +284,132 @@ class ScaleXTest : public CxxTest::TestSuite {
     return scale.getProperty("OutputWorkspace");
   }
 
+  void testTimeOfFlightAtHistogramIndex(
+      size_t i, Mantid::DataObjects::EventWorkspace_sptr inputWS,
+      boost::shared_ptr<Mantid::API::IEventWorkspace> resultEventWS,
+      double factor) {
+    auto &inEvents = inputWS->getSpectrum(i);
+    auto &outEvents = resultEventWS->getSpectrum(i);
+    TS_ASSERT_EQUALS(outEvents.getNumberEvents(), inEvents.getNumberEvents());
+
+    auto inTOFs = inEvents.getTofs();
+    auto outTOFs = outEvents.getTofs();
+    TS_ASSERT_EQUALS(inTOFs.size(), outTOFs.size());
+
+    for (size_t j = 0; j < inTOFs.size(); ++j) {
+      TS_ASSERT_DELTA(outTOFs[j], factor * inTOFs[j], 1e-12);
+    }
+  }
+
   void testScaleFactorApplied(
       const Mantid::API::MatrixWorkspace_const_sptr &inputWS,
       const Mantid::API::MatrixWorkspace_const_sptr &outputWS, double factor,
       bool multiply) {
-    const size_t xsize = outputWS->blocksize();
+
     for (size_t i = 0; i < outputWS->getNumberHistograms(); ++i) {
+      testScaleFactorAppliedAtHistIndex(i, inputWS, outputWS, factor, multiply);
+    }
+  }
+
+  ///
+  /**
+  Only loops through the bins of the index of the parameter histogram
+  The idea is to be called from within a loop that goes through the histograms
+  like on line 113
+
+  @param i :: Specifies the histogram index
+  */
+  void testScaleFactorAppliedAtHistIndex(
+      size_t i, const Mantid::API::MatrixWorkspace_const_sptr &inputWS,
+      const Mantid::API::MatrixWorkspace_const_sptr &outputWS, double factor,
+      bool multiply) {
+
+    // get bin sizes, outputWS and inputWS should be equal
+    const size_t xsize = outputWS->blocksize();
+    TS_ASSERT_EQUALS(inputWS->blocksize(), xsize);
+
+    // get all histograms of input and output
+    auto &outX = outputWS->x(i);
+    auto &outY = outputWS->y(i);
+    auto &outE = outputWS->e(i);
+
+    auto &inX = inputWS->x(i);
+    auto &inY = inputWS->y(i);
+    auto &inE = inputWS->e(i);
+
+    if (factor > 0) { // taken out of the tight loop
+      if (multiply) { // taken out of the tight loop
```

Comment:
If you leave `testScaleFactorApplied` as a special case. You can reduce the complexity of this method.